Test "tst_h_atts3.c" fails with HDF5 v1.12.0

"make check" failes when netcdf-c 4.8.0 is compiled against hdf5 1.12.0 at test "tst_h_atts3.c".
This can be fixed bij replacing line 212 in tst_h_atts3.c:

> H5O_info_t obj_info;

by:

> #if H5_VERSION_GE(1,12,0)
>       H5O_info2_t obj_info;
> #else
>       H5O_info_t obj_info;
> #endif

Greetings, Richard

I’ve also having this test failing, and I’ve seen this warning:
```
/build/netcdf/src/netcdf-c-4.8.0/h5_test/tst_h_atts3.c: In function ‘main’:
/build/netcdf/src/netcdf-c-4.8.0/h5_test/tst_h_atts3.c:282:37: warning: passing argument 6 of ‘H5Oget_info_by_idx3’ from incompatible pointer type [-Wincompatible-pointer-types]
  282 |                                  i, &obj_info, H5O_INFO_BASIC, H5P_DEFAULT) < 0) ERR;
      |                                     ^~~~~~~~~
      |                                     |
      |                                     H5O_info1_t *
In file included from /usr/include/H5Apublic.h:22,
                 from /usr/include/hdf5.h:23,
                 from /build/netcdf/src/netcdf-c-4.8.0/h5_test/tst_h_atts3.c:14:
/usr/include/H5Opublic.h:191:73: note: expected ‘H5O_info2_t *’ but argument is of type ‘H5O_info1_t *’
  191 |     H5_index_t idx_type, H5_iter_order_t order, hsize_t n, H5O_info2_t *oinfo,
      |                                                            ~~~~~~~~~~~~~^~~~~
```
Did your change fixed the test or not in the end? Or only the warning maybe?

My error output is `*** Checking vlen of compound file...Sorry! Unexpected result, /build/netcdf/src/netcdf-c-4.8.0/h5_test/tst_h_atts3.c, line: 289`

--- a/h5_test/tst_h_atts3.c
+++ b/h5_test/tst_h_atts3.c
@@ -209,7 +209,11 @@ main()
       hid_t file_typeid1[NUM_OBJ_1], native_typeid1[NUM_OBJ_1];
       hid_t file_typeid2, native_typeid2;
       hsize_t num_obj;
+#if H5_VERSION_GE(1,12,0)
+      H5O_info2_t obj_info;
+#else
       H5O_info_t obj_info;
+#endif
       char obj_name[STR_LEN + 1];
       hsize_t dims[1] = {ATT_LEN}; /* netcdf attributes always 1-D. */
 

